packages/collection-model/lib/model.js

```diff
@@ -268,12 +269,31 @@ const CollectionModel = AmpersandModel.extend(debounceActions(['fetch']), {
       // We don't care if it fails to get stats from server for any reason
     }
 
+    let isSearchIndexesSupported = false;
+    /**
+     * The support for search indexes is a feature of a server not a collection.
+     * As this check can only be performed currently by running $listSearchIndexes
+     * aggregation stage against a collection, so we run it from the collection model.
+     * With this setup, when a user opens the first collection, we set this property
+     * on the instance model and then from there its value is read avoiding call to server.
+     */
+    try {
+      isSearchIndexesSupported = await getParentByType(this, 'Instance')
+        .getIsSearchSupported({
+          dataService,
+          ns: this.ns,
+        });
+    } catch (e) {
```

Comment:
Why this try/catch? How would this fail?

Reply:
both of these would not fail as such. but if it even decides to fail, i don't want this check to break the app as the methods calling this function do not have error handling.

Reply:
We probably shouldn't ignore errors like that, if this starts unexpectedly throwing for any reason, we want this to be a signal that we see through a failing test (or worst case scenario user report)

Reply:
removed in [3ed908d](https://github.com/mongodb-js/compass/pull/4815/commits/3ed908db719f5342d3f1d7354fbdd93b5aa11119)